app/dao/notifications_dao.py

```diff
@@ -64,10 +64,44 @@ def update_job_sent_count(notification):
 
 
 def dao_update_notification(notification):
+    notification.updated_at = datetime.utcnow()
     db.session.add(notification)
     db.session.commit()
 
 
+def update_notification_status_by_id(notification_id, status):
```

Comment:
Repetitive update methods - to avoid serialising objects for speed/memory reasons.